### PR TITLE
Pre-parsing punishments in the actionbar

### PR DESCRIPTION
**Pull request by UwUAroze**

### CODE REVIEW

File: src/main/kotlin/com/github/silhouettemc/listener/player/PlayerTabCompleteListener.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+    private fun Player.sendReasonAndDurationActionBar(type: PunishmentType, target: String, unparsed: String) {
+        val parsed = PunishArgumentParser(unparsed)
+
+        val reason= parsed.reason ?: "No reason"
+        val duration = if (parsed.duration == null) "Forever" else TimeFormatter(parsed.duration!!).prettify()
+        val flags = joinFlags(parsed)
+
+        val placeholders = mapOf(
+            "actioning" to type.doingName,
```
Comment (flowergardn) [inline src/main/kotlin/com/github/silhouettemc/listener/player/PlayerTabCompleteListener.kt]: Ideally we should keep placeholder names the same across all "actions". People should be able to assume what a placeholder is based on previous messages. For example, {action} is used in error messages and broadcasts to determine what action is being issued (ban, mute, kick, etc), so this should also be that for consistency.

File: src/main/kotlin/com/github/silhouettemc/listener/player/PlayerTabCompleteListener.kt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            "duration" to duration,
+            "flags" to flags,
+            "reason" to reason,
+            "target" to target
```
Comment (flowergardn) [inline src/main/kotlin/com/github/silhouettemc/listener/player/PlayerTabCompleteListener.kt:50]: Rename this to "player" as there's no other players here. see this for more info: https://github.com/SilhouetteMC/Silhouette/pull/13#discussion_r1536833310

**Review by flowergardn — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/github/silhouettemc/listener/player/PlayerTabCompleteListener.kt (hunk 2 above)
Comment (UwUAroze) [reply src/main/kotlin/com/github/silhouettemc/listener/player/PlayerTabCompleteListener.kt:50]: Renamed all `player` to `target` instead, 93e86910afc179057b390fe534d67d87cd288087

File: src/main/kotlin/com/github/silhouettemc/listener/player/PlayerTabCompleteListener.kt (hunk 1 above)
Comment (UwUAroze) [reply src/main/kotlin/com/github/silhouettemc/listener/player/PlayerTabCompleteListener.kt]: Yeah that makes sense. Done, 90a060e16a866f08a09d01f5c14d02473495d4d8

**Review by flowergardn — APPROVED** [review]

**Pull request merged**